django/test/testcases.py

```diff
@@ -303,7 +303,10 @@ def assertRedirects(self, response, expected_url, status_code=302,
                 path = urljoin(response.request['PATH_INFO'], path)
 
             if fetch_redirect_response:
-                if netloc:
+                # netloc might be empty, or in cases where Django tests the HTTP scheme,
```

Comment:
`response.client._base_environ()['SERVER_NAME']` would be better but it might be a bit overkill.